sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosVectorIndexSpec.java

```diff
@@ -66,6 +74,127 @@ public CosmosVectorIndexSpec setType(String type) {
         checkNotNull(type, "cosmosVectorIndexType cannot be null");
         this.type = type;
         this.jsonSerializable.set(Constants.Properties.VECTOR_INDEX_TYPE, this.type, CosmosItemSerializer.DEFAULT_SERIALIZER);
+
+        // Reset restricted properties if the type is not DiskANN or QuantizedFlat
+        if (!type.equals(CosmosVectorIndexType.DISK_ANN.toString()) && !type.equals(CosmosVectorIndexType.QUANTIZED_FLAT.toString())) {
+            this.quantizationByteSize = null;
+            this.vectorIndexShardKey = null;
+        }
+        if (!type.equals(CosmosVectorIndexType.DISK_ANN.toString())) {
+            this.indexingSearchListSize = null;
+        }
+        return this;
+    }
+
+    /**
+     * Gets the quantization byte size
+     *
+     * @return quantizationByteSize the number of bytes used in product quantization of the vectors.
+     * A larger value may result in better recall for vector searches at the expense of latency.
+     * This applies to index types DiskANN and quantizedFlat. The allowed range for this parameter
+     * is between 1 and 3.
+     */
+    public Integer getQuantizationByteSize() {
+        if (this.quantizationByteSize == null && validateIndexType(false)) {
+            this.quantizationByteSize = this.jsonSerializable.getInt(Constants.Properties.VECTOR_QUANTIZATION_BYTE_SIZE);
+
+            if (this.quantizationByteSize <= 0) {
+                this.quantizationByteSize = null;
+            }
+        }
+        return this.quantizationByteSize;
+    }
+
+    private Boolean validateIndexType(boolean isIndexingSearchListSize) {
```

Comment:
I think this would be clearer as something like:

```suggestion
    private Boolean validatePropertySupportedByIndexType(IndexProperty property) {
```

along with
```
enum IndexProperty {
    SearchListSize,
    QuantizationByteSize,
    ShardKey,
}
```